cinema/views.py

```diff
@@ -173,3 +178,23 @@
 
     def perform_create(self, serializer):
         serializer.save(user=self.request.user)
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer  # Assuming MovieSerializer exists
+
+    @action(detail=True, methods=['POST'], url_path='upload-image', permission_classes=[IsAdminUser])
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = MovieImageSerializer(movie, data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
The `upload_image` action in the `MovieViewSet` is correctly implemented, but ensure that the `MovieImageSerializer` is properly defined in the serializers file and that it handles image uploads as expected.